azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -247,13 +247,45 @@ def list(self, endpoint_type: Optional[str] = None):
         elif EndpointType.AzureStorageContainer.value == endpoint_type:
             return endpoints.storage_containers
 
-    def delete(self, endpoint_name: Optional[str] = None, endpoint_type: Optional[str] = None):
+    def delete(
+        self,
+        endpoint_name: Optional[str] = None,
+        endpoint_type: Optional[str] = None,
+        force: bool = False
+    ):
         endpoints = self.hub_resource.properties.routing.endpoints
         if endpoint_type:
             endpoint_type = endpoint_type.lower()
             if EndpointType.CosmosDBContainer.value == endpoint_type and not self.support_cosmos:
                 raise InvalidArgumentValueError(INVALID_CLI_CORE_FOR_COSMOS)
 
+        if force and (self.hub_resource.properties.routing.enrichments or self.hub_resource.properties.routing.routes):
+            # collect endpoints to remove
+            endpoint_names = []
+            if endpoint_name:
+                endpoint_names.append(endpoint_name)
+            else:
+                if not endpoint_type or endpoint_type == EndpointType.EventHub.value:
+                    endpoint_names.extend([e.name for e in endpoints.event_hubs])
+                if not endpoint_type or endpoint_type == EndpointType.ServiceBusQueue.value:
+                    endpoint_names.extend([e.name for e in endpoints.service_bus_queues])
+                if not endpoint_type or endpoint_type == EndpointType.ServiceBusTopic.value:
+                    endpoint_names.extend([e.name for e in endpoints.service_bus_topics])
+                if self.support_cosmos and not endpoint_type or endpoint_type == EndpointType.CosmosDBContainer.value:
+                    endpoint_names.extend([e.name for e in endpoints.cosmos_db_sql_collections])
+                if not endpoint_type or endpoint_type == EndpointType.AzureStorageContainer.value:
+                    endpoint_names.extend([e.name for e in endpoints.storage_containers])
+            # remove enrichments
+            if self.hub_resource.properties.routing.enrichments:
+                enrichments = self.hub_resource.properties.routing.enrichments
+                enrichments = [e for e in enrichments if e.endpoint_names[0] not in endpoint_names]
+                self.hub_resource.properties.routing.enrichments = enrichments
+            # remove routes
+            if self.hub_resource.properties.routing.routes:
+                routes = self.hub_resource.properties.routing.routes
+                routes = [r for r in routes if r.endpoint_names[0] not in endpoint_names]
+                self.hub_resource.properties.routing.routes = routes
+
         if endpoint_name:
```

Comment:
I saw this in the task requirement: "On the other hand, if `--force` is not provided and associated message routes or enrichments exist - we can either prompt the user to do so and/or (where same may apply if the use declines) fail gracefully with clear instructions to remediate the issue."

therefore do we still need to add an option to prompt the user if they want to delete associated routes and message enrichments?

Reply:
good q. I do prefer warning/error over prompts so will add that

Reply:
![image](https://user-images.githubusercontent.com/73560279/234702480-3de6c6fa-051a-4c5f-9284-d0270964bd69.png)